dolfin_adjoint_common/blocks/assembly.py

```diff
@@ -18,6 +18,41 @@ def __init__(self, form, ad_block_tag=None):
     def __str__(self):
         return f"assemble({ufl2unicode(self.form)})"
 
+    def compute_action_adjoint(self, adj_input, arity_form, form=None, c_rep=None, space=None, dform=None):
```

Comment:
Can we have a docstring please? I know we don't always but this one is at the complicated end so future poor sods who have to read this will thank us for at least saying what this does and what the arguments mean.